src/typescript/13-utility-types/utility-types.md

```diff
@@ -1,117 +1,207 @@
+
+### Verify
+
+✏️ Create **src/utility-types/required.test.ts** and update it to be:
+
+@sourceref ../../../exercises/typescript/13-utility-types/01-problem/src/required.test.ts
+
+Run the following to verify your solution:
+
+```shell
+npm run test
 ```
 
-## Readonly&lt;Type&gt;
+### Exercise
+
+Update the file `required.ts` to create a new `Tyrannosaurus` type that enforces diet property to be `Diet.Carnivore` and all `Dinosaur` properties are required. Remember to take advantage of utility types and use the existing `Dinosaur` Type when creating the new `Tyrannosaurus` Type.
+
+<strong>Have issues with your local setup?</strong> You can use either [StackBlitz](https://stackblitz.com/fork/github/bitovi/academy/tree/main/exercises/typescript/13-utility-types/01-problem?file=src/required.ts) or [CodeSandbox](https://codesandbox.io/p/devbox/github/bitovi/academy/tree/main/exercises/typescript/13-utility-types/01-problem?file=src/required.ts) to do this exercise in an online code editor.
+
+### Solution
+
+<details>
+<summary>Click to see the solution</summary>
+
+✏️ Update `required.ts` to the following:
+
+@sourceref ../../../exercises/typescript/13-utility-types/01-solution/src/required.ts
+@highlight 13
+
+<strong>Have issues with your local setup?</strong> You can use either [StackBlitz](https://stackblitz.com/fork/github/bitovi/academy/tree/main/exercises/typescript/13-utility-types/01-solution?file=src/required.ts) or [CodeSandbox](https://codesandbox.io/p/devbox/github/bitovi/academy/tree/main/exercises/typescript/13-utility-types/01-solution?file=src/required.ts) to do this exercise in an online code editor.
+</details>
+
+## Objective 4: Readonly&lt;Type&gt;
 
 All properties of `Type`, but they are readonly. Use it to prevent objects from being mutated.
 
 ```typescript
+enum Diet {
+  "Carnivore",
+  "Herbivore",
+  "Omnivore",
+};
+
+interface Dinosaur {
+  species: string;
+  diet: Diet;
+  age?: number; // the question mark signals that this property is optional
+};
+
 type NamableDinosaur = { name: string } & Dinosaur; // this is an intersection between { name: string } and Dinosaur. Think { name: string } + Dinosaur
-type AnotherDinosaur = Readonly<NamableDinosaur>;
+type ReadOnlyDinosaur = Readonly<NamableDinosaur>;
 
-// Meet Bruno, another dinosaur
-let dino: AnotherDinosaur = {
+// Meet Bruno, read-only dinosaur
+const dino: ReadOnlyDinosaur = {
   name: "Bruno",
   age: 27,
   species: "Tyrannosaurus rex",
   diet: Diet.Carnivore,
 };
-dino.age = 0;
 
 // Today is its birthday! Let`s attempt to increase its age:
 dino.age += 1;
-// Cannot assign to 'age' because it is a read-only property.
 // Oops! TypeScript error
 ```
+@highlight 14, 17, 25, only
+
+In the code above, we are declaring `dino` to type `ReadOnlyDinosaur`. This will prevent us from assigning a new value because it is a read-only object.
 
-## Record&lt;Keys,Type&gt;
+## Objective 5: Record&lt;Keys, Type&gt;
```

Comment:
This can stay its own Objective and map to exercise 3 in the fall training: https://github.com/bitovi/trainings/blob/main/typescript-utility-types/Exercise3/solution/Solution.ts